informer/informer.go

```diff
@@ -0,0 +1,112 @@
+package informer
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/giantswarm/operatorkit/tpr"
+
+	microerror "github.com/giantswarm/microkit/error"
+	micrologger "github.com/giantswarm/microkit/logger"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Config represents the configuration used to create a new informer.
+type Config struct {
+	// Dependencies.
+	K8sClient            kubernetes.Interface
+	Logger               micrologger.Logger
+	ResourceEventHandler cache.ResourceEventHandler
+	TPR                  *tpr.TPR
+	ZeroObjectFactory    ZeroObjectFactory
+
+	// Settings.
+	ResyncPeriod time.Duration
+}
+
+// DefaultConfig provides a default configuration to create a new create service
+// by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		K8sClient:            nil,
+		Logger:               nil,
+		ResourceEventHandler: nil,
+		TPR:                  nil,
+		ZeroObjectFactory:    nil,
+
+		// Settings.
+		ResyncPeriod: tpr.ResyncPeriod,
+	}
+}
+
+// New returns a new configured informer.
+func New(config Config) (*cache.Controller, error) {
```

Comment:
I have strong feelings. This method needs only ResourceEventHandler, and ZeroObjectFactory - I think these should be passed as method parameters. Everything else is in TPR.